packages/web-ui-registration/src/components/LoginTerms.tsx

```diff
@@ -1,14 +1,20 @@
-import { Trans } from 'react-i18next';
 import type { ReactElement } from 'react';
-import { Link, HorizontalWizardLayoutCaption } from '@rocket.chat/layout';
+import { HorizontalWizardLayoutCaption } from '@rocket.chat/layout';
+import { useSetting } from '@rocket.chat/ui-contexts';
+import { Box } from '@rocket.chat/fuselage';
+import { useTranslation } from 'react-i18next';
 
 export const LoginTerms = (): ReactElement => {
+	const { t } = useTranslation();
+	const loginTerms = useSetting('Layout_Login_Terms') as string;
+	const loginTermsDefault =
+		'By proceeding you are agreeing to our <a href="terms-of-service">Terms of Service</a>, <a href="privacy-policy">Privacy Policy</a> and <a href="legal-notice">Legal Notice</a>.';
+
 	return (
 		<HorizontalWizardLayoutCaption>
-			<Trans
-				defaults='By proceeding you are agreeing to our <0>{{terms}}</0>, <1>{{privacyPolicy}}</1> and <2>{{legalNotice}}</2>.'
-				values={{ terms: 'Terms of Service', privacyPolicy: 'Privacy Policy', legalNotice: 'Legal Notice' }}
-				components={[<Link href='/terms-of-service' />, <Link href='/privacy-policy' />, <Link href='/legal-notice' />]}
+			<Box
+				withRichContent
+				dangerouslySetInnerHTML={{ __html: loginTerms !== loginTermsDefault ? loginTerms : t('Layout_Login_Terms_Content') }}
```

Comment:
can we not just set some value as default and handle that instead of this comparison? This strict string comparison is a bit weird considering the end result could be the same but with different formatting, rendering this bit of code useless in its intent.

the end effect would also be weird imo. if the value is **exactly** as the default considered here, if I have a server with portuguese as the server lang, the rendered content will be in portuguese, even though the setting has it in english? one space after <a and before href, and suddenly translation is gone. This would be a nighmare to troubleshoot too for those unaware of this piece of code. 

what if we leave the default empty, and if empty, run it by the default content catering to different locales. If somebody really wants to disable it, they can put an empty div there. Any other content would be exactly rendered as it is in the setting.

Reply:
in this case, I think we can let older ones stay the way they are. And just have this behaviour for the new one. So removing the default value without migration.

so workspaces with the setting set, will show what's in the setting. if they want locale support with the default content, remove the value.

We can add it to the description to make it clear?

IMO it's also worth checking the pre-react-rewrite behaviour.